RIOT-OS/RIOT
# random: add hwrng and puf_sram as FEATURES_OPTIONAL

### Contribution description

This PR adds `periph_hwrng` and `puf_sram` as `FEATURES_OPTIONAL` for `random` module.

Right now if `auto_init` is used `random` seed is initiated from `puf_sram`, `periph_hwrng`, `periph_cpuid` or `RANDOM_SEED_DEFAULT` https://github.com/RIOT-OS/RIOT/blob/999fffdc595b048c785b36a7c694d5d529c8293e/sys/random/random.c#L42-L61.

But `pub_sram` is never compiled which means cpu/boards as `samr21-xpro` end up using there CPU_ID instead of a random seed. So adding it as a `FEATURES_OPTIONAL` allows it to be included in the build if available.

### Testing procedure

- Compile `tests/puf_sram` on a board supporting `puf_sram`:

`make -C tests/puf_sram BOARD=samr21-xpro flash test`

- Enable `DEBUG` in `random.c` flash an application using random on `BOARD` providing `puf_sram` and `periph_hwrng`. In the first  case the seed should change everytime the board is power-cycled, on the second case every-time the board is reset.

<details><summary>make -C examples/gcoap/ BOARD=samr21-xpro flash term
</summary>

```
make: Entering directory '/home/francisco/workspace/RIOT/examples/gcoap'
/home/francisco/workspace/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200"
Twisted not available, please install it if you want to use pyterm's JSON capabilities
2019-09-04 09:13:41,526 - INFO # Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
2019-09-04 09:13:47,659 - INFO # random: PUF SEED not fresh
2019-09-04 09:13:47,662 - INFO # random: using seed value 973387205
2019-09-04 09:13:47,670 - INFO # main(): This is RIOT! (Version: 2019.07-devel-754-ge5de1-HEAD)
2019-09-04 09:13:47,671 - INFO # gcoap example app
2019-09-04 09:13:47,674 - INFO # All up, running the shell now
> 2019-09-04 09:13:52,725 - INFO #  random: PUF SEED not fresh
2019-09-04 09:13:52,729 - INFO # random: using seed value 2016792513
2019-09-04 09:13:52,737 - INFO # main(): This is RIOT! (Version: 2019.07-devel-754-ge5de1-HEAD)
2019-09-04 09:13:52,739 - INFO # gcoap example app
2019-09-04 09:13:52,741 - INFO # All up, running the shell now
> 2019-09-04 09:14:01,152 - INFO #  random: using seed value 2902826692
2019-09-04 09:14:01,160 - INFO # main(): This is RIOT! (Version: 2019.07-devel-754-ge5de1-HEAD)
2019-09-04 09:14:01,162 - INFO # gcoap example app
2019-09-04 09:14:01,164 - INFO # All up, running the shell now
```
</details>

<details><summary>make -C examples/gcoap/ BOARD=pba-d-01-kw2x flash -j3 term
</summary>

```
2019-09-04 09:09:14,641 - INFO # random: using seed value 3994478280
2019-09-04 09:09:14,650 - INFO # main(): This is RIOT! (Version: 2019.07-devel-754-ge5de1-HEAD)
2019-09-04 09:09:14,652 - INFO # gcoap example app
2019-09-04 09:09:14,654 - INFO # All up, running the shell now
> 2019-09-04 09:09:16,023 - INFO #  random: using seed value 3830827191
2019-09-04 09:09:16,031 - INFO # main(): This is RIOT! (Version: 2019.07-devel-754-ge5de1-HEAD)
2019-09-04 09:09:16,033 - INFO # gcoap example app
2019-09-04 09:09:16,036 - INFO # All up, running the shell now
> 2019-09-04 09:09:17,053 - INFO #  random: using seed value 920084802
2019-09-04 09:09:17,062 - INFO # main(): This is RIOT! (Version: 2019.07-devel-754-ge5de1-HEAD)
2019-09-04 09:09:17,063 - INFO # gcoap example app
2019-09-04 09:09:17,065 - INFO # All up, running the shell now
```
</details>

### Issues/PRs references
